### PR TITLE
refactor: Simplify temp file creation in tests

### PR DESCRIPTION
**Pull request by alexandear**
The PR simplifies the `openTestFile` function logic by leveraging [`T.TempDir`](https://pkg.go.dev/testing#T.TempDir).

### CODE REVIEW

File: github/github_test.go
Comment (gmlewis) [inline github/github_test.go:74]:
```suggestion
	t.Helper()
	fname := filepath.Join(t.TempDir(), name)
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @alexandear !
Now that you are passing `t` here, what do you think about making this a `t.Helper`?

File: github/github_test.go
Comment (alexandear) [reply github/github_test.go:75]: added `t.Helper()`

**Review by gmlewis — APPROVED** [review]
Thank you, @alexandear .
LGTM.  (In the future, please don't force-push in this repo as we always squash-and-merge anyway... it makes reviewing changes easier since the last review. Thanks.)

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by tomfeigin — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thanks you, @tomfeigin !
Merging.

**Pull request merged**
